pkg/graph/resolvers/system_intake_document.go

```diff
@@ -84,9 +99,115 @@ func CreateSystemIntakeDocument(ctx context.Context, store *storage.Store, s3Cli
 	return store.CreateSystemIntakeDocument(ctx, &documentDatabaseRecord)
 }
 
-// DeleteSystemIntakeDocument deletes an existing SystemIntakeDocumet, given its ID.
+// DeleteSystemIntakeDocument deletes an existing SystemIntakeDocument, given its ID.
 //
 // Does *not* delete the uploaded file from S3, following the example of TRB request documents.
 func DeleteSystemIntakeDocument(ctx context.Context, store *storage.Store, id uuid.UUID) (*models.SystemIntakeDocument, error) {
+	if err := canDelete(ctx, store, id); err != nil {
+		return nil, err
+	}
+
 	return store.DeleteSystemIntakeDocument(ctx, id)
 }
+
+// canView guards unauthorized views (downloads) of system intake documents
+// admins can view any document
+// requesters can view any document
+// GRB reviewers can view any document
+func canView(ctx context.Context, store *storage.Store, s3Key string) error {
+	// admins can view
+	if services.HasRole(ctx, models.RoleEasiGovteam) {
+		return nil
+	}
+
+	document, err := store.GetSystemIntakeDocumentByS3Key(ctx, s3Key)
+	if err != nil {
+
+		if errors.Is(err, sql.ErrNoRows) {
+			// if the document was deleted, don't error and break
+			return nil
+		}
```

Comment:
@adamodd if you were curious what that issue was. it wasn't on the delete at all. finally realized this morning that it was err no rows on the `URL` resolver for a system intake document, and that runs after deletion. in `main`, we go straight to S3 for that URL, but in this auth patch, we first check the DB to see who created the document, but it is gone after deletion!